src/main/java/uk/gov/justice/digital/domain/DomainExecutor.java

```diff
@@ -0,0 +1,271 @@
+        if (domainOperation.equalsIgnoreCase("insert")) {
+            storage.create(tablePath, dataFrame);
+        } else if (domainOperation.equalsIgnoreCase("update")) {
+            if (storage.exists(info)) {
+                storage.replace(tablePath, dataFrame);
+            } else {
+                throw new DomainExecutorException("Delta table " + info.getTable() + "doesn't exists");
+            }
+        } else if (domainOperation.equalsIgnoreCase("sync")) {
+            if (storage.exists(info)) {
+                storage.reload(tablePath, dataFrame);
+            } else {
+                throw new DomainExecutorException("Delta table " + info.getTable() + "doesn't exists");
+            }
+        } else {
+            // Handle invalid operation type
+            logger.error("Invalid operation type " + domainOperation);
+            throw new DomainExecutorException("Invalid operation type " + domainOperation);
+        }
+        storage.endTableUpdates(info);
+        storage.vacuum(info);
+    }
+
+    @Override
+    protected void saveViolations(final TableInfo target, final Dataset<Row> dataFrame) {
+        String tablePath = storage.getTablePath(target.getPrefix(), target.getSchema(), target.getTable());
+        // save the violations to the specified location
+        storage.append(tablePath, dataFrame);
+        storage.endTableUpdates(target);
+    }
+
+    public void deleteFull(final TableInfo info) throws DomainExecutorException {
+        logger.info("DomainOperations:: deleteFull");
+        if (storage.exists(info)) {
+            storage.delete(info);
+            storage.endTableUpdates(info);
+            storage.vacuum(info);
+        } else {
+            throw new DomainExecutorException("Table " + info.getTable() + "doesn't exists");
+        }
+    }
+
+    public Dataset<Row> getAllSourcesForTable(final String sourcePath, final String source,
+                                              final TableTuple exclude) throws DomainExecutorException {
+        if(exclude != null && exclude.asString().equalsIgnoreCase(source)) {
+            //TODO: this condition only for unit test
+            // we already have this table
+            logger.info("table already present " + exclude.asString());
+        } else {
+            try {
+                TableTuple full = new TableTuple(source);
+                final Dataset<Row> dataFrame = storage.load(
+                        TableInfo.create(sourcePath, full.getSchema(), full.getTable()));
+                if(dataFrame != null) {
+                    logger.info("Loaded source '" + full.asString() +"'.");
+                    return dataFrame;
+                } else {
+                    logger.error("Source " + full.asString() + " not found");
+                    throw new DomainExecutorException("Source " + full.asString() + " not found");
+                }
+            } catch(Exception e) {
+                handleError(e);
```

Comment:
Handle error just calls `logger.error` so remove `handleError` and use `logger.error` here directly as per the comments on my previous review.

Reply:
done